lib/internal/modules/esm/loader.js

```diff
@@ -225,6 +231,39 @@ class ESMLoader {
     };
   }
 
+  /**
+   * Returns the url to use for the resolution of a given cache key url
+   * These are not guaranteed to be the same.
+   *
+   * In WHATWG HTTP spec for ESM the cache key is the non-I/O bound
+   * synchronous resolution using only string operations
+   *   ~= resolveImportMap(new URL(specifier, importerHREF))
+   *
+   * The url used for subsequent resolution is the response URL after
+   * all redirects have been resolved.
+   *
+   * https://example.com/foo redirecting to https://example.com/bar
+   * would have a cache key of https://example.com/foo and baseURL
+   * of https://example.com/bar
+   *
+   * MUST BE SYNCHRONOUS for import.meta initialization
+   * MUST BE CALLED AFTER receiving the url body due to I/O
+   * @param {string} url
+   * @returns {string}
+   */
+  baseURL(url) {
```

Comment:
Is `baseURL` the official name for this? (I'm wondering if it's related to browser-land's `document.baseURI`). If not, I'm not crazy about the name—I think it's too ambiguous and makes me think it will behave like `basename()`.

Perhaps `getResolvedURL`? (adding the `get` prefix to associate with a accessor/`Map::get()` operation rather than an input manipulation like `basename()`)

Reply:
yup, that's the name for better or worse see quotes in https://github.com/nodejs/node/pull/42131#issue-1150651498

Reply:
eh, none of these functions are pure so they all don't do just input manipulation. unclear what makes this one different